app/components/OceanInterface/OceanInterface.tsx

```diff
@@ -21,61 +23,78 @@ async function handlePress (txid: string): Promise<void> {
   }
 }
 
-async function broadcastTransaction (tx: OceanTransaction, client: WhaleApiClient, setError: (e: Error) => void): Promise<string | undefined> {
-  let retries = 0
+async function broadcastTransaction (tx: OceanTransaction, client: WhaleApiClient, retries: number = 0): Promise<string> {
   try {
     return await client.transactions.send({ hex: tx.signed.toHex() })
   } catch (e) {
-    retries++
     Logging.error(e)
-    if (retries < 1) {
-      return await broadcastTransaction(tx, client, setError)
+    if (retries < MAX_AUTO_RETRY) {
+      return await broadcastTransaction(tx, client, retries + 1)
     }
-    setError(e)
+    throw e
   }
 }
 
 /**
  * @description - Global component to be used for async calls, network errors etc. This component is positioned above the bottom tab.
  * Need to get the height of bottom tab via `useBottomTabBarHeight()` hook to be called on screen.
  * */
-export function OceanInterface (): JSX.Element {
-  const { height, err: e } = useSelector((state: RootState) => state.ocean)
-  const transaction = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
-  const [tx, setTx] = useState<OceanTransaction>(transaction)
+export function OceanInterface (): JSX.Element | null {
   const dispatch = useDispatch()
   const client = useWhaleApiClient()
+
+  // store
+  const { height, err: e } = useSelector((state: RootState) => state.ocean)
+  const transaction = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
+
+  // state
+  const [tx, setTx] = useState<OceanTransaction|undefined>(transaction)
   const [err, setError] = useState<Error | undefined>(e)
+
   const slideAnim = useRef(new Animated.Value(0)).current
   Animated.timing(slideAnim, { toValue: height, duration: 300, useNativeDriver: false }).start()
 
+  const dismissDrawer = useCallback(() => {
+    setTx(undefined)
+    setError(undefined)
+  }, [])
+
   useEffect(() => {
+    // last available job will remained in this UI state until get dismissed
     if (transaction !== undefined) {
-      broadcastTransaction(transaction, client, setError).then(() => {
-        setTx({
-          ...transaction,
-          title: translate('screens/OceanInterface', 'Transaction complete'),
-          broadcasted: true
-        })
-        dispatch(ocean.actions.popTransaction())
-      }).catch(() => {
+      setTx({
+        ...transaction,
+        title: translate('screens/OceanInterface', 'Transaction complete'),
+        broadcasted: false
       })
+
+      broadcastTransaction(transaction, client)
+        .catch((e: Error) => setError(e))
+        .finally(() => dispatch(ocean.actions.popTransaction())) // remove the job as soon as completion
     }
   }, [transaction])
 
+  if (tx === undefined) {
```

Comment:
`tx` is the ONE and ONLY must have state to render the component. Logic/hooks before this line should control it properly.